pynbody/halo/adaptahop.py

```diff
@@ -320,14 +352,23 @@ def _can_load(cls, sim, arr_name="grp", *args, **kwa):
         if len(valid_candidates) == 0:
             return False
 
-        longint = kwa.pop("longint", False)
+        # Logic is as follows:
+        # - If `longint` is provided, try loading with it.
+        # - Otherwise, try loading with and without it
+        use_longint = kwa.pop("longint", None)
+        if use_longint is None:
+            longint_flags = [True, False]
+        else:
+            longint_flags = use_longint
         for fname in valid_candidates:
             with FortranFile(fname) as fpu:
```

Comment:
This block of code seems like a repetition of the above format detection. Any chance the longint test could be broken into a smaller function to avoid this?

Reply:
Should be better now!

Reply:
Looks perfect!